### ISSUE TITLE
State is not maintained when hot reloading

### ISSUE THREAD

**Issue by Dundercover** [FrimJo/react-take-off#2]
If you change anything, the app will do a hard reload and reset to initial state. This is a bit problematic if you are just changing some styles for some component and have to get back to where you were after every change to witness the result. Can we solve this somehow? Could it be typescript that is recompiling everything? Do we need to set `transpileOnly: true`?

**Comment by FrimJo** [comment]
Thank you dundercover for taking an interest in this small project. I do not believe it’s related to type script, but I’ll have a look during the day. If you come up with a solution, feel free to create a PR.

**Comment by FrimJo** [comment]
@Dundercover I have pushed a new version to master, containing changes to almost fix hot-reloading issue.

I added a temporary component in `App.tsx` to test hot reloading. One of the problems were that `tslint --fix` automatically added function calls (which where using `dispatch` or `useCallback` under the hood) to varying dependencies arrays for hooks further up the tree. This resulted in unnecessary re-renders. A dirty solution was to add `// eslint-disable-next-line react-hooks/exhaustive-deps` to all these places, though I believe there is a more "correct" way of fixing this issue, all feedback are welcome.

Still though, the fetch component re-renders when hot-reloading, couldn't figure that on out. Will probably have a new look at it tomorrow.
